src/svg/qsvggenerator.cpp

```diff
@@ -388,7 +395,7 @@ class QSvgPaintEngine : public QPaintEngine
         case Qt::CustomDashLine: {
             QString color, colorOpacity, dashPattern, dashOffset;
 
-            qreal penWidth = spen.width() == 0 ? qreal(1) : spen.widthF();
+            qreal penWidth = spen.widthF() == 0 ? qreal(1) : spen.widthF();
```

Comment:
seems like you are now comparing float to integer

Reply:
My book says that the integer type will now be changed to float before they're compared, but I can change it if you want.

Reply:
type wise yes. but comparing float to zero can be tricky (see e.g. https://bitbashing.io/comparing-floats.html)